### DISCUSSION TITLE
why methods are deprecated Nette\Utils::getReturnTypes(),  Nette\Utils::getParameterTypes() Nette\Utils::getPropertyTypes() ?

### DISCUSSION THREAD

**Discussion by petrparolek** [nette/utils#267] [General]
Versions: 3.1.6 and 3.2.5

### Bug Description
Methods are deprecated Nette\Utils::getReturnTypes(),  Nette\Utils::getParameterTypes() Nette\Utils::getPropertyTypes() ?

### Expected Behavior
Not deprecated these methods

### Description
I found these methods are still used in Nette packages

**Comment by dg** [comment]
Because in PHP 8.1 there are new intersection types that it can't handle. So instead of this, there is Type class.

https://doc.nette.org/type

**Comment by petrparolek** [comment]
@dg Are you planning make refactoring code depends `Nette\Utils::getReturnTypes()`, `Nette\Utils::getParameterTypes()` and `Nette\Utils::getPropertyTypes()` in Nette 3.0 and 3.1?

**Reply by dg** [reply]
I dont understand

**Comment by petrparolek** [comment]
OK, česky - když jsou metody Nette\Utils::getReturnTypes(), Nette\Utils::getParameterTypes() and Nette\Utils::getPropertyTypes() označeny jako deprecated, plánuješ refactoring kodu Nette 3.0 a 3.1, který volá tyto metody?

**Reply by dg** [reply]
Už by nikde neměly být pouzivane
